forms/viewsets/form.py

```diff
@@ -109,6 +117,49 @@ def get_serializer_class(self):
         return super().get_serializer_class()
 
 
+class AnswerPublicViewSet(AnswerViewSet):
+    """Public API for submitting form answers with added restrictions."""
+
+    http_method_names = ["post"]  # In public API, only POST is allowed
+    serializer_class = AnswerPublicSerializer
+    filter_backends = None
+    filterset_class = None
+
+    def get_queryset(self):
+        return super(AnswerViewSet, self).get_queryset()
+
+    def get_serializer_class(self):
+        return super(AnswerViewSet, self).get_serializer_class()
+
+    def create(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        answer_id = response.data.get("id")
```

Comment:
Actully also considering. lines 136 - 152 are not doing "create". I would maybe consider if it would be meaningful to create own method for this.

Reply:
It has been refactored